tests/testthat/test-matrix.r

```diff
@@ -288,13 +292,14 @@ test_that("nonsensical operations are warned about", {
 })
 
 test_that("warning is thrown if it is assumed that the survey is representative", {
-  expect_warning(
-    contact_matrix(
+  expect_snapshot(
+    # nolint start
```

Comment:
do we need this because `tmp` is not used and, if yes, do we even need it?

Reply:
tmp isn't used, it's a bit of a workaround to avoid the snapshot capturing the entire output of the table, but we only want the warning.

We want to capture the warning message, and it seems that changing over to the `cli` warning messages slightly break the regex that `expect_warning` was testing for. 

Ideally we would use `expect_snapshot_warning` but it seems it is in a questioning lifecycle (https://testthat.r-lib.org/reference/expect_snapshot_output.html), so if I use `expect_snapshot()` on the output, it captures the whole table. We could get around this by using `expect_snapshot_warning()` and then eventually if the testthat deprecates the function we can deal with it then. I have asked them about this here: https://github.com/r-lib/testthat/issues/2182.

Reply:
Overall I think it is cleaner to just use `expect_snapshot_warning()`, so I'll do that, I think.

Reply:
It might be worthwhile thinking about using snapshot testing in other parts of socialmixr however, as it is a bit more complete than using regexes. I'll post an issue about it.

Reply:
Using expect_snapshot_warning() resolves this :)